app/models/manageiq/providers/vmware/inventory/collector/network_manager.rb

```diff
@@ -0,0 +1,152 @@
+class ManageIQ::Providers::Vmware::Inventory::Collector::NetworkManager < ManageIQ::Providers::Vmware::Inventory::Collector
+  attr_reader :network_name_mapping
+  VappNetwork = Struct.new(:id, :name, :type, :is_shared, :gateway, :dns1, :dns2, :netmask, :enabled, :dhcp_enabled)
+
+  def orgs
+    return @orgs if @orgs.any?
+    @orgs = connection.organizations
+  end
+
+  def vdc_networks
+    return @vdc_networks if @vdc_networks.any?
+    @vdc_networks = orgs.each_with_object([]) do |org, res|
+      res.concat(org.networks.all)
+    end
+  end
+
+  def vapp_networks
+    return @vapp_networks if @vapp_networks.any?
+    @vapp_networks = manager.orchestration_stacks.each_with_object([]) do |stack, res|
+      fetch_network_configurations_for_vapp(stack.ems_ref).map do |net_conf|
+        # 'none' is special network placeholder that we must ignore
+        next if net_conf[:networkName] == 'none'
+
+        $vcloud_log.debug("#{log_header} processing net_conf for vapp #{stack.ems_ref}: #{net_conf}")
```

Comment:
We should update fog gem to do this kind of low-level processing for us. But we haven't yet, so leave it here 😇